test/dlc_tests/sanity/test_pre_release.py

```diff
@@ -0,0 +1,218 @@
+import re
+
+import pytest
+
+from invoke.context import Context
+
+from test.test_utils import LOGGER, ec2
+
+
+def test_stray_files(image):
+    """
+    Test to ensure that unnecessary build artifacts are not present in any easily visible or tmp directories
+
+    :param image: ECR image URI
+    """
+    ctx = Context()
+    container_name = f"test_tmp_dirs-{image.split('/')[-1].replace('.', '-').replace(':', '-')}"
+    _start_container(container_name, image, ctx)
+
+    # Running list of artifacts/artifact regular expressions we do not want in any of the directories
+    stray_artifacts = [r"\.py"]
+
+    # Ensure stray artifacts are not in the tmp directory
+    tmp = _run_cmd_on_container(container_name, ctx, "ls -A /tmp")
+    _assert_artifact_free(tmp, stray_artifacts)
+
+    # Ensure tmp dir is empty except for whitelisted files
+    allowed_tmp_files = ["hsperfdata_root", ".", ".."]
```

Comment:
note: i can get rid of "." and ".." since we are using ls -a, but figured i'd be safe here